# Preços de Medicamentos - Consumidor
### Nesse projeto irei criar uma API para saber de fato de quanto foi o aumento no preço dos médicamento nesse último reajuste no brasil.

<img src="https://i.imgur.com/W7DMmnl.jpeg" />

Os dados representam a lista de preços de Medicamentos , contemplando o preço Fábrica, ou preço **fabricante (PF)**, que é o preço máximo que pode ser praticado pelas empresas produtoras ou importadoras do produto e pelas empresas distribuidoras. O PF indica o preço máximo permitido para venda a farmácias e drogarias e o **Preço Máximo ao Consumidor (PMC)** indica o preço teto de venda ao consumidor.

Fonte dos dados: https://dados.anvisa.gov.br/dados

### Agência Nacional de Vigilância Sanitária - ANVISA

Criada pela Lei nº 9.782, de 26 de janeiro 1999, a Agência Nacional de Vigilância Sanitária (Anvisa) é uma autarquia sob regime especial, que tem sede e foro no Distrito Federal, e está presente em todo o território nacional por meio das coordenações de portos, aeroportos, fronteiras e recintos alfandegados.

Tem por finalidade institucional promover a proteção da saúde da população, por intermédio do controle sanitário da produção e consumo de produtos e serviços submetidos à vigilância sanitária, inclusive dos ambientes, dos processos, dos insumos e das tecnologias a eles relacionados, bem como o controle de portos, aeroportos, fronteiras e recintos alfandegados.

- Data da criação do banco de dados: 31 de maio de 2017
- Penúltima atualização: 13 de outubro de 2020 para o DataFrame anterior
- Ultima atualização: 08 de fevereiro de 2021 para o DataFrame reajuste

In [1]:
import pandas as pd
import numpy as np
import scipy
from scipy import stats
import matplotlib.pyplot as plt
import seaborn as sns
import folium
%matplotlib inline



import warnings
warnings.filterwarnings("ignore")
pd.set_option('display.max_columns', 999)

In [2]:
import mysql.connector
import pandas as a

conn = mysql.connector.connect(host="localhost", user="root", passwd="1234", db="preco_medicamentos_brasil")
anterior = a.read_sql_query("SELECT * FROM TA_PRECO_MEDICAMENTO_10_2020" 
                      ,conn)


reajuste = a.read_sql_query("SELECT * FROM TA_PRECO_MEDICAMENTO_03_2021" 
                      ,conn)

In [3]:
anterior.head()

SUBSTANCIA                CNPJ  \
0  ACETATO DE DEXAMETASONA;CLOTRIMAZOL  18.459.628/0001-15   
1                           ABATACEPTE  56.998.982/0001-07   
2                           ABATACEPTE  56.998.982/0001-07   
3                           ABATACEPTE  56.998.982/0001-07   
4                           ABCIXIMABE  51.780.468/0001-87   

                              LABORATORIO CODIGO GGREM  REGISTRO     EAN 1  \
0                              BAYER S.A.     5,39E+14  1,71E+12  7,89E+12   
1  BRISTOL-MYERS SQUIBB FARMACĘUTICA LTDA     5,05E+14  1,02E+12  7,90E+12   
2  BRISTOL-MYERS SQUIBB FARMACĘUTICA LTDA     5,05E+14  1,02E+12  7,90E+12   
3  BRISTOL-MYERS SQUIBB FARMACĘUTICA LTDA     5,05E+14  1,02E+12  7,90E+12   
4         JANSSEN-CILAG FARMACĘUTICA LTDA     5,15E+14  1,12E+12  7,90E+12   

        EAN 2       EAN 3     PRODUTO  \
0      -           -       BAYCUTEN N   
1      -           -          ORENCIA   
2      -           -          ORENCIA   
3      -           -          ORENCIA   
4      -           -           REOPRO   

                                        APRESENTACAO  \
0     10 MG/G + 0,443 MG/G CREM DERM CT BG AL X 40 G   
1              125 MG/ML SOL INJ CT 1 SER PREENCHIDA   
2  125 MG/ML SOL INJ CT 4 SER PREENC VD TRANS + D...   
3     250 MG PO LIOF INJ 1 CT 1 FA + SER DESCARTÁVEL   
4                2 MG/ML SOL INJ CT FA VD INC X 5 ML   

                                  CLASSE TERAPEUTICA  \
0  D7B2 - CORTICOESTERÓIDES ASSOCIADOS A ANTIMICO...   
1          M1C - AGENTES ANTI-REUMÁTICOS ESPECÍFICOS   
2          M1C - AGENTES ANTI-REUMÁTICOS ESPECÍFICOS   
3          M1C - AGENTES ANTI-REUMÁTICOS ESPECÍFICOS   
4  B1C3 - INIBIDORES DA AGREGAÇĂO PLAQUETÁRIA, AN...   

  TIPO DE PRODUTO (STATUS DO PRODUTO) REGIME DE PRECO PF Sem Impostos  \
0                                Novo        Regulado           19,74   
1                           Biológico        Regulado         1228,49   
2                           Biológico        Regulado         4914,04   
3                           Biológico        Regulado         1547,45   
4                              -             Regulado         2003,06   

     PF 0%   PF 12%   PF 17% PF 17% ALC PF 17,5% PF 17,5% ALC   PF 18%  \
0    22,11    25,55    27,32      23,78    27,51        23,93    27,71   
1  1228,49  1396,01  1480,11    1480,11  1489,08      1489,08  1498,16   
2  4914,04  5584,14  5920,53    5920,53  5956,41      5956,41  5992,73   
3  1547,45  1758,47   1864,4     1864,4   1875,7       1875,7  1887,13   
4  2003,06  2276,21  2413,32    2413,32  2427,95      2427,95  2442,76   

  PF 18% ALC   PF 20%   PMC 0%  PMC 12%  PMC 17% PMC 17% ALC PMC 17,5%  \
0      27,71    28,51    29,66    34,13    36,42       32,87     36,66   
1    1498,16  1535,61  1698,32   1929,9  2046,17     2046,17   2058,57   
2    5992,73  6142,55  6793,37  7719,75  8184,79     8184,79   8234,39   
3    1887,13  1934,31                                                    
4    2442,76  2503,83                                                    

  PMC 17,5% ALC  PMC 18% PMC 18% ALC  PMC 20% RESTRICAO HOSPITALAR  CAP  \
0         33,08    36,92       33,28    37,95                  Năo  Năo   
1       2058,57  2071,12     2071,12  2122,89                  Năo  Năo   
2       8234,39   8284,6      8284,6  8491,72                  Năo  Năo   
3                                                              Sim  Năo   
4                                                              Sim  Năo   

  CONFAZ 87 ICMS 0% ANALISE RECURSAL  \
0       Năo     Năo                    
1       Năo     Năo                    
2       Sim     Năo                    
3       Năo     Năo                    
4       Năo     Năo                    

  LISTA DE CONCESSAO DE CREDITO TRIBUTARIO (PIS/COFINS) COMERCIALIZACAO 2019  \
0                                           Negativa                     Sim   
1                                           Positiva                     Năo   
2   

In [4]:
reajuste.head()

SUBSTANCIA (REAJUSTE)     CNPJ (REAJUSTE)  \
0  ACETATO DE LANREOTIDA  07.718.721/0001-80   
1              ACICLOVIR  45.992.062/0001-65   
2              ACICLOVIR  45.992.062/0001-65   
3              ACICLOVIR  45.992.062/0001-65   
4     ACTAEA RACEMOSA L.  78.950.011/0001-20   

                LABORATORIO (REAJUSTE) CODIGO GGREM (REAJUSTE)  \
0     BEAUFOUR IPSEN FARMACÊUTICA LTDA                5,39E+14   
1             GERMED FARMACEUTICA LTDA                5,26E+14   
2             GERMED FARMACEUTICA LTDA                5,26E+14   
3             GERMED FARMACEUTICA LTDA                5,26E+14   
4  HERBARIUM LABORATORIO BOTANICO LTDA                5,11E+14   

  REGISTRO (REAJUSTE) EAN 1 (REAJUSTE) EAN 2 (REAJUSTE) EAN 3 (REAJUSTE)  \
0            1,70E+12         7,90E+12         7,90E+12           -        
1            1,06E+12         7,90E+12           -                -        
2            1,06E+12         7,90E+12           -                -        
3            1,06E+12         7,90E+12           -                -        
4            1,19E+12         7,90E+12           -                -        

   PRODUTO (REAJUSTE)                            APRESENTACAO (REAJUSTE)  \
0  SOMATULINE AUTOGEL  120 MG SOL INJ LIB PROL CT ENV AL/PLAS SER PRE...   
1           ACICLOVIR             400 MG COM CT BL AL PLAS PVDC OPC X 20   
2           ACICLOVIR             400 MG COM CT BL AL PLAS PVDC OPC X 30   
3           ACICLOVIR             400 MG COM CT BL AL PLAS PVDC OPC X 50   
4            CLIFEMIN       160 MG COM REV CT BL AL PLAS PVDC TRANS X 15   

                       CLASSE TERAPEUTICA (REAJUSTE)  \
0                   H1C2 - HORMÔNIOS ANTICRESCIMENTO   
1                      J5B3 - ANTIVIRAIS PARA HERPES   
2                      J5B3 - ANTIVIRAIS PARA HERPES   
3                      J5B3 - ANTIVIRAIS PARA HERPES   
4  G3J - MODULADORES SELETIVOS DO RECEPTOR DE EST...   

  TIPO DE PRODUTO (STATUS DO PRODUTO) (REAJUSTE) REGIME DE PRECO (REAJUSTE)  \
0                                           Novo                   Regulado   
1                                       Genérico                   Regulado   
2                                       Genérico                   Regulado   
3                                       Genérico                   Regulado   
4                                   Fitoterápico                   Liberado   

  PF Sem Impostos (REAJUSTE) PF 0% (REAJUSTE) PF 12% (REAJUSTE)  \
0                    2456,35          2456,35           2791,31   
1                      57,69            57,69             65,56   
2                      86,54            86,54             98,34   
3                     144,23           144,23             163,9   
4                       34,1            38,19             44,13   

  PF 17% (REAJUSTE) PF 17% ALC (REAJUSTE) PF 17,5% (REAJUSTE)  \
0           2959,46               2959,46             2977,39   
1             69,51                 69,51               69,93   
2            104,27                104,27               104,9   
3            173,77                173,77              174,82   
4             47,19                 41,08               47,52   

  PF 17,5% ALC (REAJUSTE) PF 18% (REAJUSTE) PF 18% ALC (REAJUSTE)  \
0                 2977,39           2995,55               2995,55   
1                   69,93             70,35                 70,35   
2                   104,9            105,54                105,54   
3                  174,82            175,89                175,89   
4                   41,33             47,86                 41,58   

  PF 20% (REAJUSTE) PMC 0% (REAJUSTE) PMC 12% (REAJUSTE) PMC 17% (REAJUSTE)  \
0           3070,44           3395,76            3858,82            4091,28   
1             72,11             79,75              90,63              96,09   
2            108,18            119,64             135,95             144,15   
3            180,29            199,39             226,58          

In [5]:
anterior = anterior.astype(str)
reajuste = reajuste.astype(str)

In [6]:
anterior.rename(columns = lambda x: x.replace(' ', '_'), inplace=True)
reajuste.rename(columns = lambda x: x.replace(' ', '_'), inplace=True)

In [7]:
anterior.replace({',': '.'}, regex=True, inplace=True)
reajuste.replace({',': '.'}, regex=True, inplace=True)

In [9]:
anterior['TARJA'] = anterior['TARJA'].apply(lambda x: x.replace('(*)',''))
reajuste['TARJA_(REAJUSTE)'] = reajuste['TARJA_(REAJUSTE)'].apply(lambda x: x.replace('(*)',''))

**Temos aqui as bases das últimas atualizações de medicamento no país, para inicia nossa EDA vamos verificar de quanto foi o aumento dos médicamentos.**

## Dicionário das variáveis 

**ANÁLISE RECURSAL** – Destina-se a prestar informações sobre produtos cuja análise de
preço ainda esteja em curso no âmbito da CMED, tanto em sede de pedido de reconsideração
como de recurso ao Comitê Técnico Executivo - CTE/CMED.

**APRESENTAÇÃO** – É a descrição de como o medicamento é apresentado, quanto a sua
forma farmacêutica, dosagem e quantidade.

**CAP** – O Coeficiente de Adequação de Preços é um desconto mínimo obrigatório,
atualizado anualmente, a ser aplicado sempre que forem realizadas vendas de medicamentos,
constantes do rol divulgado pela CMED e nas compras de todos os medicamentos por força de
decisão judicial, destinadas aos entes da administração pública direta e indireta da União, dos
Estados, do Distrito Federal e dos Municípios. Sobre ele, vamos falar mais à frente.

**CLASSE TERAPÊUTICA** – É a Classificação Anatômica de Produtos Farmacêuticos. Está
relacionado com a finalidade e o local de atuação do medicamento de acordo com a Associação
Europeia de Pesquisa de Mercado Farmacêutico (EphMRA).

**CNPJ** – Código de identificação da pessoa jurídica. 

**CÓDIGO GGREM** – É um campo específico de identificação de determinada
apresentação, cujo fornecimento e geração são exclusivos da CMED e são únicos para cada
apresentação.

**COMERCIALIZAÇÃO 2017** – Refere-se à informação se o produto foi ou não
comercializado pelo detentor do registro, no ano de 2017.

**CONFAZ 87** – Lista de medicamentos que ficam isentas do ICMS, cujas operações sejam
destinadas a órgãos da Administração Pública Direta e Indireta Federal, Estadual e Municipal e
do Distrito Federal e a suas fundações públicas.

**EAN 1** – O Código EAN é também conhecido como Código de Barras ou GTIN. É composto
por 13 dígitos e nele se identifica o País de origem do produto, a empresa fabricante, o produto
por ela produzido e um digito verificador.

**EAN 2** – O Código EAN é também conhecido como Código de Barras ou GTIN. É composto
por 13 dígitos e nele se identifica o País de origem do produto, a empresa fabricante, o produto
por ela produzido e um digito verificador.

**EAN 3** – O Código EAN é também conhecido como Código de Barras ou GTIN. É composto
por 13 dígitos e nele se identifica o País de origem do produto, a empresa fabricante, o produto
por ela produzido e um digito verificador.

**LABORATÓRIO** – É o nome da empresa detentora do registro sanitário podendo ser a
produtora ou importadora do medicamento.

**LISTA DE CONCESSÃO DE CRÉDITO TRIBUTÁRIO (PIS/COFINS)** – Classificação fiscal dos
medicamentos em uma das listas a seguir, conforme o regime tributário previsto na Lei nº
10.147, de 2000:

- **Lista Positiva:** medicamentos cujas substâncias ativas constam do anexo do Decreto nº 3.803, de 24 de abril de 2001, e suas atualizações, sujeitos à prescrição médica, identificados com tarja vermelha ou preta, e cujas empresas produtoras usufruem do regime especial de utilização de crédito presumido de PIS/Pasep e COFINS, classificados em três categorias, ou seja, medicamentos isentos de PIS/Pasep e COFINS:
    - I.Categoria I - Medicamentos monodroga identificados com tarja vermelha ou preta; 
    - II.Categoria II - Medicamentos em associações identificados com tarja vermelha ou preta;
    - III.Categoria III - Substâncias para medicamentos utilizados em nutrição parenteral, hemodiálise e diálise peritoneal, substitutos do plasma e       expansores plasmáticos, identificados com tarja vermelha. 
- **Lista Negativa:** medicamentos pertencentes às classificações constantes do art. 1º daLei nº 10.147, de 2000, cujas substâncias ativas não estejam relacionadas no anexo do Decreto nº 3.803, de 2001, e suas atualizações, ou seja, medicamentos incidência de 12% de PIS/Pasep e COFINS.
- **Lista Neutra:** medicamentos que não estão sujeitos ao regime tributário estabelecidona Lei nº 10.147, de 2000, ou seja, medicamentos incidência de 9,25% de PIS/Pasep e COFINS.

**PF** - PF 0%, PF 12%, PF 17%, PF 17% ALC, PF 17,5%, PF 17,5% ALC, PF 18%, PF 18% ALC, PF
20%, PF Sem Impostos, Preço Fábrica – É o teto de preço pelo qual um laboratório ou
distribuidor de medicamentos pode comercializar no mercado brasileiro um medicamento.
Desta maneira, o PF vem a ser o preço máximo permitido para vendas de medicamentos
destinadas a farmácias, drogarias, além das destinadas a entes da Administração Pública.

**PMC** - PMC 0%, PMC 12%, PMC 17%, PMC 17% ALC, PMC 17,5%, PMC 17,5% ALC, PMC 18%,
PMC 18% ALC, PMC 20%, Preço Máximo ao Consumidor ou PMC – É o preço a ser praticado pelo
comércio varejista de medicamentos, ou seja, farmácias e drogarias, tendo em vista que este
contempla tanto a margem de lucro como os impostos inerentes a esses tipos de comércio.

*A Alíquota de 0% se refere ao preço sem ICMS, que deve ser aplicado aos medicamentos isentos
de ICMS, que são aqueles relacionados nos Convênios do CONFAZ, por exemplo CONVÊNIO ICMS
162/94 que isenta medicamentos destinados ao tratamento de câncer.*

*A Alíquota de ICMS 20% se refere atualmente apenas ao Estado do RJ.*

*A Alíquota de ICMS 18% se refere aos Estados de AM, AP, BA, CE, MA, MG, PB, PE, PI, PR, RN,
RS, SE, SP, TO.*

*A Alíquota de ICMS 17,5% se refere ao Estado de RO*

*A alíquota de ICMS 17% está relacionada aos demais Estados*

*A Alíquota de ICMS 12% está relacionada com os Medicamentos Genéricos de SP e MG.*

*As Alíquotas de 17%, 17,5% e 18% que tem ao lado a sigla ALC, significa Áreas de Livre Comércio,
e dizem respeito às seguintes cidades: Manaus/Tabatinga (AM), Boa Vista/Bonfim (RR),
Macapá/Santana (AP), Guajará-Mirim (RO), Brasiléia/Epitaciolândia/ Cruzeiro do Sul (AC).*

*Além das alíquotas citadas anteriormente também é publicado o PF Sem Impostos que são os
preços sem o ICMS e sem PIS/Pasep e COFINS.*

**PRINCÍPIO ATIVO** – É uma substância química ativa, fármaco, droga ou matéria-prima
que tenha propriedades farmacológicas com finalidade medicamentosa, utilizada para
diagnóstico, alívio ou tratamento, empregada para modificar ou explorar sistemas fisiológicos
ou estados patológicos, em benefício da pessoa na qual se administra. Portanto, princípio ativo
do medicamento é o nome principal de sua composição.

**PRODUTO** – É o nome comercial dado ao medicamento, em determinada apresentação,
de um referido laboratório, no caso dos medicamentos genéricos é igual ao Princípio Ativo.

**REGISTRO** – É o número de registro de produto junto à ANVISA.

**RESTRIÇÃO HOSPITALAR** – Indica os medicamentos em embalagens hospitalares e de
uso restrito a hospitais e clínicas e, portanto, não podem ser comercializados pelo Preço Máximo
ao Consumidor não tendo seus PMC publicados.

**TARJA** – Indica qual se a venda do medicamento necessita de recita, sendo classificado em: 

- **Tarja Vermelha:** Medicamentos que só podem ser dispensados mediante apresentação de receita.
- **Tarja Preta:** Medicamentos que devem ser dispensados apenas mediante receituário acompanhado de notificação de receita, devendo ficar uma cópia no estabelecimento que realizou a dispensação.
- **Venda livre:** Medicamentos que podem ser dispensados sem a apresentação de receita.

**TIPO DE PRODUTO (STATUS DO PRODUTO)** – Tipo é a categorização do medicamento
por tipo de produto em Biológicos, Biológico Novo, Similar, Genérico, Novo, Específico,
Radiofármaco.

Status do produto indica se ele é referência.

**ICMS 0%** – Indica se o medicamento tem ou não o imposto sobre Circulação de
Mercadorias e Prestação de Serviços (ICMS), onde o imposto de cada Estado deve ser
considerado, pois ele influencia na diferença dos preços dos medicamentos de um Estado para
o outro. Por exemplo, no Estado do Paraná o ICMS é de 18% (dezoito por cento), no Estado do
Rio de Janeiro é de 20% (vinte por cento) e no Estado do Goiás é de 17% (dezessete por cento).

**Vamos unir os dataframes para comparação.**

In [10]:
df_merge = reajuste.merge(anterior, how = 'left', left_on = ['CNPJ_(REAJUSTE)', 'LABORATORIO_(REAJUSTE)', 'PRODUTO_(REAJUSTE)', 'APRESENTACAO_(REAJUSTE)', 'TARJA_(REAJUSTE)'], right_on = ['CNPJ', 'LABORATORIO', 'PRODUTO', 'APRESENTACAO', 'TARJA'])

In [11]:
df_merge['PF_Sem_Impostos_(REAJUSTE)'] = df_merge['PF_Sem_Impostos_(REAJUSTE)'].astype('float')
df_merge['PF_Sem_Impostos'] = df_merge['PF_Sem_Impostos'].astype('float')

In [12]:
df_merge

SUBSTANCIA_(REAJUSTE)     CNPJ_(REAJUSTE)  \
0      ACETATO DE LANREOTIDA  07.718.721/0001-80   
1                  ACICLOVIR  45.992.062/0001-65   
2                  ACICLOVIR  45.992.062/0001-65   
3                  ACICLOVIR  45.992.062/0001-65   
4         ACTAEA RACEMOSA L.  78.950.011/0001-20   
...                      ...                 ...   
26002              ZOPICLONA  02.685.377/0001-57   
26003          ZUCLOPENTIXOL  04.522.600/0001-70   
26004          ZUCLOPENTIXOL  04.522.600/0001-70   
26005          ZUCLOPENTIXOL  04.522.600/0001-70   
26006          ZUCLOPENTIXOL  04.522.600/0001-70   

                    LABORATORIO_(REAJUSTE) CODIGO_GGREM_(REAJUSTE)  \
0         BEAUFOUR IPSEN FARMACÊUTICA LTDA                5.39E+14   
1                 GERMED FARMACEUTICA LTDA                5.26E+14   
2                 GERMED FARMACEUTICA LTDA                5.26E+14   
3                 GERMED FARMACEUTICA LTDA                5.26E+14   
4      HERBARIUM LABORATORIO BOTANICO LTDA                5.11E+14   
...                                    ...                     ...   
26002     SANOFI-AVENTIS FARMACÊUTICA LTDA                5.03E+14   
26003                 LUNDBECK BRASIL LTDA                5.24E+14   
26004                 LUNDBECK BRASIL LTDA                5.24E+14   
26005                 LUNDBECK BRASIL LTDA                5.24E+14   
26006                 LUNDBECK BRASIL LTDA                5.24E+14   

      REGISTRO_(REAJUSTE) EAN_1_(REAJUSTE) EAN_2_(REAJUSTE) EAN_3_(REAJUSTE)  \
0                1.70E+12         7.90E+12         7.90E+12           -        
1                1.06E+12         7.90E+12           -                -        
2                1.06E+12         7.90E+12           -                -        
3                1.06E+12         7.90E+12           -                -        
4                1.19E+12         7.90E+12           -                -        
...                   ...              ...              ...              ...   
26002            1.13E+12         7.90E+12           -                -        
26003            1.05E+12         5.70E+12           -                -        
26004            1.05E+12         5.70E+12           -                -        
26005            1.05E+12         5.70E+12           -                -        
26006            1.05E+12         5.70E+12           -                -        

       PRODUTO_(REAJUSTE)                            APRESENTACAO_(REAJUSTE)  \
0      SOMATULINE AUTOGEL  120 MG SOL INJ LIB PROL CT ENV AL/PLAS SER PRE...   
1               ACICLOVIR             400 MG COM CT BL AL PLAS PVDC OPC X 20   
2               ACICLOVIR             400 MG COM CT BL AL PLAS PVDC OPC X 30   
3               ACICLOVIR             400 MG COM CT BL AL PLAS PVDC OPC X 50   
4                CLIFEMIN       160 MG COM REV CT BL AL PLAS PVDC TRANS X 15   
...                   ...                                                ...   
26002             IMOVANE            7.5 MG COM REV CT BL AL PLAS TRANS X 20   
26003            CLOPIXOL            10 MG COM REV CT  BL AL PLAS TRANS X 20   
26004            CLOPIXOL             25 MG COM REV CT BL AL PLAS TRANS X 20   
26005            CLOPIXOL   50 MG/ML SOL INJ ACUPHASE CT AMP VD TRANS X 1 ML   
26006            CLOPIXOL  200 MG/ML SOL INJ IM DEPOT CX AMP VD TRANS X 1 ML   

                           CLASSE_TERAPEUTICA_(REAJUSTE)  \
0                       H1C2 - HORMÔNIOS ANTICRESCIMENTO   
1                          J5B3 - ANTIVIRAIS PARA HERPES   
2                          J5B3 - ANTIVIRAIS PARA HERPES   
3                          J5B3 - ANTIVIRAIS PARA HERPES   
4      G3J - MODULADORES SELETIVOS DO RECEPTOR DE EST...   
...                                                  ...   
26002  N5B1 - HIPNÓTICOS E SEDATIVOS NÃO BARBITÚRICOS...   
26003                N5A9 - ANTIPSICÓTICOS CONVENCIONAIS   
26004                N5A9 - ANTIPSICÓTICOS CONVENCIONAIS   
26005              

**Agora vamos fazer uma comparação entre o valor anterior com o reajuste**

In [13]:
for index, row in df_merge.iterrows():
    if row['APRESENTACAO_(REAJUSTE)'] == row['APRESENTACAO'] and row['PRODUTO_(REAJUSTE)'] == row['PRODUTO'] and row['LABORATORIO_(REAJUSTE)'] == row['LABORATORIO']:
        df_merge.loc[index,'VALOR_REAJUSTE_%'] =  100*((row['PF_Sem_Impostos_(REAJUSTE)']/row['PF_Sem_Impostos'])- 1)

In [14]:
verificacao = df_merge[['SUBSTANCIA_(REAJUSTE)','LABORATORIO_(REAJUSTE)', 'PRODUTO_(REAJUSTE)', 'APRESENTACAO_(REAJUSTE)', 'SUBSTANCIA', 'LABORATORIO', 'PRODUTO', 'APRESENTACAO', 'PF_Sem_Impostos', 'PF_Sem_Impostos_(REAJUSTE)', 'VALOR_REAJUSTE_%']]
verificacao

SUBSTANCIA_(REAJUSTE)               LABORATORIO_(REAJUSTE)  \
0      ACETATO DE LANREOTIDA     BEAUFOUR IPSEN FARMACÊUTICA LTDA   
1                  ACICLOVIR             GERMED FARMACEUTICA LTDA   
2                  ACICLOVIR             GERMED FARMACEUTICA LTDA   
3                  ACICLOVIR             GERMED FARMACEUTICA LTDA   
4         ACTAEA RACEMOSA L.  HERBARIUM LABORATORIO BOTANICO LTDA   
...                      ...                                  ...   
26002              ZOPICLONA     SANOFI-AVENTIS FARMACÊUTICA LTDA   
26003          ZUCLOPENTIXOL                 LUNDBECK BRASIL LTDA   
26004          ZUCLOPENTIXOL                 LUNDBECK BRASIL LTDA   
26005          ZUCLOPENTIXOL                 LUNDBECK BRASIL LTDA   
26006          ZUCLOPENTIXOL                 LUNDBECK BRASIL LTDA   

       PRODUTO_(REAJUSTE)                            APRESENTACAO_(REAJUSTE)  \
0      SOMATULINE AUTOGEL  120 MG SOL INJ LIB PROL CT ENV AL/PLAS SER PRE...   
1               ACICLOVIR             400 MG COM CT BL AL PLAS PVDC OPC X 20   
2               ACICLOVIR             400 MG COM CT BL AL PLAS PVDC OPC X 30   
3               ACICLOVIR             400 MG COM CT BL AL PLAS PVDC OPC X 50   
4                CLIFEMIN       160 MG COM REV CT BL AL PLAS PVDC TRANS X 15   
...                   ...                                                ...   
26002             IMOVANE            7.5 MG COM REV CT BL AL PLAS TRANS X 20   
26003            CLOPIXOL            10 MG COM REV CT  BL AL PLAS TRANS X 20   
26004            CLOPIXOL             25 MG COM REV CT BL AL PLAS TRANS X 20   
26005            CLOPIXOL   50 MG/ML SOL INJ ACUPHASE CT AMP VD TRANS X 1 ML   
26006            CLOPIXOL  200 MG/ML SOL INJ IM DEPOT CX AMP VD TRANS X 1 ML   

          SUBSTANCIA           LABORATORIO   PRODUTO  \
0                NaN                   NaN       NaN   
1                NaN                   NaN       NaN   
2                NaN                   NaN       NaN   
3                NaN                   NaN       NaN   
4                NaN                   NaN       NaN   
...              ...                   ...       ...   
26002            NaN                   NaN       NaN   
26003            NaN                   NaN       NaN   
26004  ZUCLOPENTIXOL  LUNDBECK BRASIL LTDA  CLOPIXOL   
26005            NaN                   NaN       NaN   
26006            NaN                   NaN       NaN   

                                 APRESENTACAO  PF_Sem_Impostos  \
0                                         NaN              NaN   
1                                         NaN              NaN   
2                                         NaN              NaN   
3                                         NaN              NaN   
4                                         NaN              NaN   
...                                       ...              ...   
26002                                     NaN              NaN   
26003                                     NaN              NaN   
26004  25 MG COM REV CT BL AL PLAS TRANS X 20            57.02   
26005                                     NaN              NaN   
26006                                     NaN              NaN   

       PF_Sem_Impostos_(REAJUSTE)  VALOR_REAJUSTE_%  
0                         2456.35               NaN  
1                           57.69               NaN  
2                           86.54               NaN  
3                          144.23               NaN  
4                           34.10               NaN  
...                           ...               ...  
26002                       35.36               NaN  
26003                       26.08               NaN  
26004                       57.02               0.0  
26005                       26.22               NaN  
26006                       52.32               NaN  

[26007 rows x 11 columns]

In [15]:
pd.set_option('display.float_format', lambda x: '%.2f' % x)

In [16]:
verificacao['VALOR_REAJUSTE_%'].value_counts()

0.00      1776
4.22         2
4.22         2
5.24         2
-4.05        2
2.99         1
25.75        1
5.19         1
-4.99        1
3.27         1
3.16         1
4.23         1
4.21         1
28.31        1
3.21         1
5.21         1
-78.37       1
37.79        1
5.21         1
5.21         1
-4.05        1
5.20         1
3.23         1
5.21         1
4.22         1
3.22         1
4.21         1
3.16         1
4.08         1
5.17         1
4.22         1
5.20         1
4.21         1
25.77        1
4.27         1
4.23         1
65.99        1
-12.46       1
3.22         1
-4.96        1
5.21         1
3.22         1
5.21         1
4.13         1
5.26         1
4.21         1
Name: VALOR_REAJUSTE_%, dtype: int64

**Temos aqui os valores em porcentagem do reajuste, tivemos 1776 medicamentos que não tiveram reajuste em seu preço.**

## Quais medicamentos tiveram aumento em seus preços ?

**Agora vamos verificar quais medicamentos tiveram aumento em seus preços olhado para o seu preço base, ou seja, sem impostos.**

In [17]:
aumento = verificacao[verificacao['VALOR_REAJUSTE_%'] > 0]

In [18]:
maior_aumento = aumento.sort_values(by='VALOR_REAJUSTE_%', ascending=False)
maior_aumento.head(25)

SUBSTANCIA_(REAJUSTE)  \
12880                           ESTRADIOL HEMI-HIDRATADO   
25399                        TRIMETOPRIMA;SULFAMETOXAZOL   
9428                            CLORIDRATO DE RANITIDINA   
9807           CLORIDRATO DE TETRACICLINA;ANFOTERICINA B   
9809           CLORIDRATO DE TETRACICLINA;ANFOTERICINA B   
17496                                        LIMECICLINA   
22111                                        PREGABALINA   
22118                                        PREGABALINA   
22117                                        PREGABALINA   
20145                                         OLANZAPINA   
2074                                         ARIPIPRAZOL   
20144                                         OLANZAPINA   
22116                                        PREGABALINA   
2345                               ATORVASTATINA CÁLCICA   
22115                                        PREGABALINA   
20203                             OLMESARTANA MEDOXOMILA   
22112                                        PREGABALINA   
20204                             OLMESARTANA MEDOXOMILA   
5871   CITRATO DE SÓDIO DIHIDRATADO;CITRATO DE POTÁSS...   
8803                             CLORIDRATO DE NEBIVOLOL   
8352                             CLORIDRATO DE MEMANTINA   
8360                             CLORIDRATO DE MEMANTINA   
8355                             CLORIDRATO DE MEMANTINA   
15971                  IBANDRONATO DE SÓDIO MONOIDRATADO   
8356                             CLORIDRATO DE MEMANTINA   

                                  LABORATORIO_(REAJUSTE)  \
12880  BESINS HEALTHCARE BRASIL COMERCIAL E DISTRIBUI...   
25399                        EUROFARMA LABORATÓRIOS S.A.   
9428                CIMED INDÚSTRIA DE MEDICAMENTOS LTDA   
9807                                   ARESE PHARMA LTDA   
9809                                   ARESE PHARMA LTDA   
17496                                            EMS S/A   
22111                             APSEN FARMACEUTICA S/A   
22118                           LABORATÓRIOS PFIZER LTDA   
22117                           LABORATÓRIOS PFIZER LTDA   
20145                                            EMS S/A   
2074                                             EMS S/A   
20144                                            EMS S/A   
22116                           LABORATÓRIOS PFIZER LTDA   
2345                            GERMED FARMACEUTICA LTDA   
22115                           LABORATÓRIOS PFIZER LTDA   
20203                        EUROFARMA LABORATÓRIOS S.A.   
22112                             APSEN FARMACEUTICA S/A   
20204                        EUROFARMA LABORATÓRIOS S.A.   
5871                     PROCTER & GAMBLE DO BRASIL LTDA   
8803                            GERMED FARMACEUTICA LTDA   
8352                                             EMS S/A   
8360                                             EMS S/A   
8355                                             EMS S/A   
15971                                            EMS S/A   
8356                                             EMS S/A   

                      PRODUTO_(REAJUSTE)  \
12880                            VAGIFEM   
25399                            SUBTRAX   
9428                           ULCEROCIN   
9807                          TERICIN AT   
9809                          TERICIN AT   
17496                        LIMECICLINA   
22111                             INSIT®   
22118                              ALOND   
22117                              ALOND   
20145                              ZOPIX   
2074                              ARPEJO   
20144                              ZOPIX   
22116                              ALOND   
2345                            ATORLESS   
22115                              ALOND   
20203                             HOLMES   
22112                             INSIT®   
20204                             HOLMES   
5871                           FLORALYTE   
8803             CLORIDRATO DE NE

**Os dois medicamentos com maior aumento foram o VAGIFEM 10 MCG COM REV VAG CT ENVOL APLIC PREENC PLAS eo SUBTRAX 80 MG/ML + 16 MG/ML SUS OR CT FR VD AMB X 100 ML**

## Quais medicamentos tiveram queda em seus preços ?

**Agora iremos verificar quais medicamentos tiveram queda em seus preços.**

In [19]:
queda = aumento = verificacao[verificacao['VALOR_REAJUSTE_%'] < 0]

In [20]:
maior_queda = queda.sort_values(by='VALOR_REAJUSTE_%', ascending=False)
maior_queda.head(25)

SUBSTANCIA_(REAJUSTE)       LABORATORIO_(REAJUSTE)  \
4900   CEFTRIAXONA DISSÓDICA HEMIEPTAIDRATADA  ANTIBIÓTICOS DO BRASIL LTDA   
4905   CEFTRIAXONA DISSÓDICA HEMIEPTAIDRATADA  ANTIBIÓTICOS DO BRASIL LTDA   
4907   CEFTRIAXONA DISSÓDICA HEMIEPTAIDRATADA  ANTIBIÓTICOS DO BRASIL LTDA   
113                               PREGABALINA                      EMS S/A   
111                               PREGABALINA                      EMS S/A   
14902                     HEDERA HELIX (HERA)             FARMOQUÍMICA S/A   
6794                 CLORIDRATO DE AMIODARONA  HIPOLABOR FARMACEUTICA LTDA   

                           PRODUTO_(REAJUSTE)  \
4900   CEFTRIAXONA DISSÓDICA HEMIEPTAIDRATADA   
4905   CEFTRIAXONA DISSÓDICA HEMIEPTAIDRATADA   
4907   CEFTRIAXONA DISSÓDICA HEMIEPTAIDRATADA   
113                                    KONDUZ   
111                                    KONDUZ   
14902                                 ABRILAR   
6794                 CLORIDRATO DE AMIODARONA   

                                 APRESENTACAO_(REAJUSTE)  \
4900           1 G PO SOL INJ CT 25 FA VD TRANS X 10 ML    
4905          1 G PO SOL INJ CT 50 FA VD TRANS X 10 ML     
4907   1 G PO SOL INJ CT 50 FA VD TRANS X 10 ML + 50 ...   
113     35.0 MG CAP DURA CT BL AL PLAS PVC/PVDC OPC X 28   
111      35.0 MG CAP DURA CT BL AL PLAS PVC/PVDC OPC X 7   
14902  7 MG/ML SOL OR CT FR VD AMB X 200 ML + COP (SA...   
6794        50 MG/ML SOL INJ IV CX 100 AMP VD AMB X 3 ML   

                                   SUBSTANCIA                  LABORATORIO  \
4900   CEFTRIAXONA DISSÓDICA HEMIEPTAIDRATADA  ANTIBIÓTICOS DO BRASIL LTDA   
4905   CEFTRIAXONA DISSÓDICA HEMIEPTAIDRATADA  ANTIBIÓTICOS DO BRASIL LTDA   
4907   CEFTRIAXONA DISSÓDICA HEMIEPTAIDRATADA  ANTIBIÓTICOS DO BRASIL LTDA   
113                               PREGABALINA                      EMS S/A   
111                               PREGABALINA                      EMS S/A   
14902                     HEDERA HELIX (HERA)             FARMOQUÍMICA S/A   
6794                 CLORIDRATO DE AMIODARONA  HIPOLABOR FARMACEUTICA LTDA   

                                      PRODUTO  \
4900   CEFTRIAXONA DISSÓDICA HEMIEPTAIDRATADA   
4905   CEFTRIAXONA DISSÓDICA HEMIEPTAIDRATADA   
4907   CEFTRIAXONA DISSÓDICA HEMIEPTAIDRATADA   
113                                    KONDUZ   
111                                    KONDUZ   
14902                                 ABRILAR   
6794                 CLORIDRATO DE AMIODARONA   

                                            APRESENTACAO  PF_Sem_Impostos  \
4900           1 G PO SOL INJ CT 25 FA VD TRANS X 10 ML            799.67   
4905          1 G PO SOL INJ CT 50 FA VD TRANS X 10 ML            1599.33   
4907   1 G PO SOL INJ CT 50 FA VD TRANS X 10 ML + 50 ...          1599.33   
113     35.0 MG CAP DURA CT BL AL PLAS PVC/PVDC OPC X 28            33.68   
111      35.0 MG CAP DURA CT BL AL PLAS PVC/PVDC OPC X 7             8.42   
14902  7 MG/ML SOL OR CT FR VD AMB X 200 ML + COP (SA...            42.70   
6794        50 MG/ML SOL INJ IV CX 100 AMP VD AMB X 3 ML           670.00   

       PF_Sem_Impostos_(REAJUSTE)  VALOR_REAJUSTE_%  
4900                       767.29             -4.05  
4905                      1534.57             -4.05  
4907                      1534.57             -4.05  
113                         32.01             -4.96  
111                          8.00             -4.99  
14902                       37.38            -12.46  
6794                       144.92            -78.37

**Os medicamentos que tiveram queda em seus preços foram poucos com destaque para o medicamento CLORIDRATO DE AMIODARONA 50 MG/ML SOL INJ IV CX 100 AMP VD AMB X 3 ML que teve um desconto de 78.37% em relação ao valor praticado no último reajuste.**

## Análisando os medicamentos usados no procedimento de intubação para paciente graves de Covid-19.

**Vinte e dois estados e o Distrito Federal estão com seus estoques de insumos para a intubação de pacientes graves de covid-19.**

**Anestésico local** 
- **Lidocaína** (2%) sem vasoconstritor: esse medicamento é um anestésico utilizado no primeiro atendimento do paciente, quando ele apresenta insuficiência respiratória.

**Ansiolítico** 
- **Diazepam**: é usado para sedação e para evitar crise convulsiva, que ocorre com pacientes em estado grave. 

**Bloqueadores neuromusculares**
- **Besilato de atracúrio, besilato de cisatracúrio e brometo de rocurônio**: esses medicamentos são importantes para facilitar a intubação e essenciais para mantê-la, porque neutralizam os movimentos respiratórios. Além disso, são utilizados em pacientes que permanecem intubados por um longo período de tempo.

**Bloqueador neuromuscular** 
- **Cloreto de suxametônio**: servem para neutralizar os movimentos respiratórios. Este tipo de bloqueador é necessário apenas na sequência rápida de intubação, isto é, no primeiro atendimento ao paciente.

**Fármaco antídoto**
- **Cloridrato de naloxona**: serve para diminuir os efeitos colaterais de outros medicamentos, como os opioides (que aliviam a dor). Se houver uma super-dosagem ou até uma intoxicação aguda pelos opioides, consequentemente, o batimento cardíaco do paciente e a respiração pode diminuir e até haver uma redução da motilidade intestinal.

**Fármaco para controle de delírio**
- **Haloperidol**: como indica sua categoria, o medicamento é usado para controle de delírio do paciente. O problema, que pode ocorrer por intoxicação medicamentosa, deve ser evitado pois pode fazer com que a pessoa apresente convulsões. Segundo os especialistas, há também pacientes que podem delirar por abstinência alcoólica.

**Fármaco para sedação contínua** 
- **Cloridrato de midazolam, propofol, e cloridrato de cetamina**: são sedativos, ou seja, têm efeito calmante e agem no controle da irritabilidade, nervosismo ou excitação. Além disso, a cetamina tem a função de prolongar o efeito da anestesia.

**Fármaco para sedação contínua e controle de delírio** 
- **Cloridrato de dexmedetomidina**: são medicamentos fortes, usados para sedação e ao mesmo tempo controle de delírio do paciente. 

**Fármaco para analgesia**
- **Cloridrato de dextrocetamina, citrato de fentanila e sulfato de morfina**: são usados para alívio das dores e em processo de cuidados paliativos em pacientes com mau prognóstico, que apresentam baixa capacidade de recuperação. São medicamentos essenciais para permanência dos pacientes na UTI.

**Fármacos vasoativos**
- **Sulfato de atropina, hermitartarato de epinefrina, hermitartarato de norepinefrina**: eles ajudam no controle do coração e na circulação pulmonar de pacientes intubados. Também são usados em casos de parada cardiorrespiratória e estados de choque, portanto, são essenciais.

**Sedativo**
- **Etomidato**: é utilizado no primeiro atendimento do paciente, quando há insuficiência respiratória.

### Sabendo desses medicamentos vamos filtra o dataset para verificar qual foi o reajuste desses medicamentos tão importante nesse momento.

In [21]:
med_intubacao = verificacao.loc[verificacao['SUBSTANCIA_(REAJUSTE)'].isin(['DIAZEPAM'])]
med_intubacao

SUBSTANCIA_(REAJUSTE)  \
11309              DIAZEPAM   
11310              DIAZEPAM   
11311              DIAZEPAM   
11312              DIAZEPAM   
11313              DIAZEPAM   
11314              DIAZEPAM   
11315              DIAZEPAM   
11316              DIAZEPAM   
11317              DIAZEPAM   
11318              DIAZEPAM   
11319              DIAZEPAM   
11320              DIAZEPAM   
11321              DIAZEPAM   
11322              DIAZEPAM   
11323              DIAZEPAM   
11324              DIAZEPAM   
11325              DIAZEPAM   
11326              DIAZEPAM   
11327              DIAZEPAM   
11328              DIAZEPAM   
11329              DIAZEPAM   
11330              DIAZEPAM   
11331              DIAZEPAM   
11332              DIAZEPAM   
11333              DIAZEPAM   
11334              DIAZEPAM   
11335              DIAZEPAM   
11336              DIAZEPAM   
11337              DIAZEPAM   
11338              DIAZEPAM   
11339              DIAZEPAM   
11340              DIAZEPAM   
11341              DIAZEPAM   
11342              DIAZEPAM   
11343              DIAZEPAM   
11344              DIAZEPAM   
11345              DIAZEPAM   
11346              DIAZEPAM   
11347              DIAZEPAM   
11348              DIAZEPAM   
11349              DIAZEPAM   
11350              DIAZEPAM   
11351              DIAZEPAM   
11352              DIAZEPAM   
11353              DIAZEPAM   
11354              DIAZEPAM   

                                  LABORATORIO_(REAJUSTE)  \
11309         LEGRAND PHARMA INDÚSTRIA FARMACÊUTICA LTDA   
11310         LEGRAND PHARMA INDÚSTRIA FARMACÊUTICA LTDA   
11311             FUNDAÇÃO PARA O REMÉDIO POPULAR - FURP   
11312             FUNDAÇÃO PARA O REMÉDIO POPULAR - FURP   
11313                LABORATÓRIO FARMACÊUTICO DA MARINHA   
11314    CRISTÁLIA PRODUTOS QUÍMICOS FARMACÊUTICOS LTDA.   
11315                PHARLAB INDÚSTRIA FARMACÊUTICA S.A.   
11316                              EMS SIGMA PHARMA LTDA   
11317                              EMS SIGMA PHARMA LTDA   
11318    BRAINFARMA INDÚSTRIA QUÍMICA E FARMACÊUTICA S.A   
11319    BRAINFARMA INDÚSTRIA QUÍMICA E FARMACÊUTICA S.A   
11320    BRAINFARMA INDÚSTRIA QUÍMICA E FARMACÊUTICA S.A   
11321    BRAINFARMA INDÚSTRIA QUÍMICA E FARMACÊUTICA S.A   
11322            UNIÃO QUÍMICA FARMACÊUTICA NACIONAL S/A   
11323            UNIÃO QUÍMICA FARMACÊUTICA NACIONAL S/A   
11324    BRAINFARMA INDÚSTRIA QUÍMICA E FARMACÊUTICA S.A   
11325    BRAINFARMA INDÚSTRIA QUÍMICA E FARMACÊUTICA S.A   
11326       PRODUTOS ROCHE QUÍMICOS E FARMACÊUTICOS S.A.   
11327       PRODUTOS ROCHE QUÍMICOS E FARMACÊUTICOS S.A.   
11328       PRODUTOS ROCHE QUÍMICOS E FARMACÊUTICOS S.A.   
11329       PRODUTOS ROCHE QUÍMICOS E FARMACÊUTICOS S.A.   
11330               SANTISA LABORATÓRIO FARMACÊUTICO S/A   
11331               SANTISA LABORATÓRIO FARMACÊUTICO S/A   
11332               SANTISA LABORATÓRIO FARMACÊUTICO S/A   
11333               SANTISA LABORATÓRIO FARMACÊUTICO S/A   
11334               SANTISA LABORATÓRIO FARMACÊUTICO S/A   
11335    CRISTÁLIA PRODUTOS QUÍMICOS FARMACÊUTICOS LTDA.   
11336    CRISTÁLIA PRODUTOS QUÍMICOS FARMACÊUTICOS LTDA.   
11337    CRISTÁLIA PRODUTOS QUÍMICOS FARMACÊUTICOS LTDA.   
11338  LABORATÓRIO NEO QUÍMICA COMÉRCIO E INDÚSTRIA LTDA   
11339  LABORATÓRIO NEO QUÍMICA COMÉRCIO E INDÚSTRIA LTDA   
11340  LABORATÓRIO NEO QUÍMICA COMÉRCIO E INDÚSTRIA LTDA   
11341  LABORATÓRIO NEO QUÍMICA COMÉRCIO E INDÚSTRIA LTDA   
11342                   LABORATÓRIO TEUTO BRASILEIRO S/A   
11343            UNIÃO QUÍMICA FARMACÊUTICA NACIONAL S/A   
11344            UNIÃO QUÍMICA FARMACÊUTICA NACIONAL S/A   
11345            UNIÃO QUÍMICA FARMACÊUTICA NACIONAL S/A   
11346                        HIPOLABOR FARMACEUTICA LTDA   
11347                              EMS SIGMA PHARMA LTDA   
11348                              EMS SIGMA PHARMA LTDA   
11349         LEGRAND PHARMA INDÚSTRIA FARMACÊUTICA LTDA   
11350         LEGRAND PHARMA